### DISCUSSION TITLE
Uncaught (in promise) Error: Failed to decrypt

### DISCUSSION THREAD

**Discussion by onchainguy-btc** [WalletConnect/walletconnect-monorepo#2731] [Developer Support]
We have issues with the migration on our rc site: https://rc-app.polychainmonsters.com

The modal properly opens but when scanning the QR code, the console will print the following error: 
```
Uncaught (in promise) Error: Failed to decrypt
    at 2982.75fc87aedf28b3dc.js:20:26514
    at iP.decode (2982.75fc87aedf28b3dc.js:20:26557)
    at o.<anonymous> (2982.75fc87aedf28b3dc.js:20:92159)
    at o.emit (_app-c3785235a5093d37.js:248:192756)
    at iW.onMessageEvent (2982.75fc87aedf28b3dc.js:20:45933)
    at async iW.onProviderPayload (2982.75fc87aedf28b3dc.js:20:45780)
```

This can be reproduced using different wallets. This is how we integrate wallet connect: 

```react
import { Chain, configureChains, createConfig, WagmiConfig } from 'wagmi';
import { mainnet, goerli, polygon, polygonMumbai, bscTestnet, bsc } from 'wagmi/chains';
import { MetaMaskConnector } from 'wagmi/connectors/metaMask';
import { InjectedConnector } from 'wagmi/connectors/injected';
import { WalletConnectConnector } from 'wagmi/connectors/walletConnect';
import { CoinbaseWalletConnector } from 'wagmi/connectors/coinbaseWallet';
import { JsonRpcProviderList, Web3ProviderTypes } from './web3-provider.types';
import { alchemyProvider } from 'wagmi/providers/alchemy';
import { publicProvider } from 'wagmi/providers/public';
import { SolanaWalletProvider } from './solana-wallet-connect';
import { generateJsonRpcProvider } from './utils';

const supportedChains = [mainnet, goerli, polygon, polygonMumbai, bsc, bscTestnet];

const ankrRpcProviderList: JsonRpcProviderList = [
  { id: mainnet.id, url: 'https://rpc.ankr.com/eth' },
  { id: goerli.id, url: 'https://rpc.ankr.com/eth_goerli' },
  { id: polygon.id, url: 'https://rpc.ankr.com/polygon' },
  { id: polygonMumbai.id, url: 'https://rpc.ankr.com/polygon_mumbai' },
  { id: bsc.id, url: 'https://rpc.ankr.com/bsc' },
  { id: bscTestnet.id, url: 'https://data-seed-prebsc-1-s1.binance.org:8545/' } // ankr unusable for bsc testnet
];

const chainlistProviderList: JsonRpcProviderList = [
  { id: mainnet.id, url: 'https://eth-rpc.gateway.pokt.network' },
  { id: goerli.id, url: 'https://eth-goerli.public.blastapi.io' },
  { id: polygon.id, url: 'https://polygon-bor.publicnode.com' },
  { id: polygonMumbai.id, url: 'https://polygon-testnet.public.blastapi.io' },
  { id: bsc.id, url: 'https://bsc-mainnet.public.blastapi.io' },
  { id: bscTestnet.id, url: 'https://data-seed-prebsc-1-s3.binance.org:8545/' }
];
const ankrProvider = generateJsonRpcProvider(ankrRpcProviderList);
const chainlistProvider = generateJsonRpcProvider(chainlistProviderList);

const apiKey = process.env.NEXT_PUBLIC_KEY_ALCHEMY as string;

const { chains, publicClient } = configureChains<Chain>(
  supportedChains,
  [...ankrProvider, ...chainlistProvider, publicProvider(), alchemyProvider({ apiKey })],
  {
    pollingInterval: 30_000,
    stallTimeout: 3000
  }
);

const config = createConfig({
  autoConnect: true,
  connectors: [
    new MetaMaskConnector({
      chains
    }),
    new WalletConnectConnector({
      chains,
      options: {
        projectId: 'xxx',
        qrModalOptions: {
          //See https://docs.walletconnect.com/2.0/web/web3modal/react/wagmi/custom-wallets#custom-explorer-wallets - Metamask, Rainbow and TrustWallet
          explorerRecommendedWalletIds: [
            'c57ca95b47569778a828d19178114f4db188b89b763c899ba0be274e97267d96',
            '1ae92b26df02f0abca6304df07debccd18262fdf5fe82daa81593582dac9a369',
            '4622a2b2d6af1c9844944291e5e7351a6aa24cd7b23099efac1b2fd875da31a0'
          ],
          chainImages: null
        }
      }
    }),
    new CoinbaseWalletConnector({
      chains,
      options: {
        appName: 'xxx'
      }
    }),
    new InjectedConnector({
      chains,
      options: {
        name: 'Injected',
        shimDisconnect: true
      }
    })
  ],
  publicClient
});

export function Web3Provider({ children }: Web3ProviderTypes) {
  return (
    <WagmiConfig config={config}>
      <SolanaWalletProvider>{children}</SolanaWalletProvider>
    </WagmiConfig>
  );
}
```

**Comment by finessevanes** [comment]
@onchainguy-eth Hey could you include your package.json? Please make sure you're using the latest versions.

**Reply by onchainguy-btc** [reply]
These are the relevant lines from my `package.json`: 

```
"viem": "^1.1.2",
"wagmi": "1.2.1",
```

`wagmi` uses these deps: 
```
"@walletconnect/ethereum-provider": "2.8.1",
"@walletconnect/legacy-provider": "^2.0.0",
"@walletconnect/modal": "^2.4.6",
```

**Reply by onchainguy-btc** [reply]
@finessevanes any idea?

**Reply by finessevanes** [reply]
@onchainguy-eth what wallet are you using to connect?

**Reply by onchainguy-btc** [reply]
@finessevanes we've tried TrustWallet, Rainbow and Metamask.

**Reply by onchainguy-btc** [reply]
On localhost everything works btw., in the same app.

**Reply by finessevanes** [reply]
Do you have a staging link we can test?

**Reply by onchainguy-btc** [reply]
@finessevanes it's rolled out on our [rc system](https://rc-app.polychainmonsters.com/)

**Reply by finessevanes** [reply]
@onchainguy-eth thanks! Will share with our team for further investigation.

**Reply by onchainguy-btc** [reply]
@finessevanes any news?

**Reply by aaronmgdr** [reply]
experiencing the same issues on https://app.stcelo.xyz/connect  and on [staked-celo-web-app-git-rainbowkit-c-labs.vercel.app](https://staked-celo-web-app-git-rainbowkit-c-labs.vercel.app/) 

One uses our react-celo package and the other uses rainbowkit. 

It was working a few weeks ago.

**Reply by dckesler** [reply]
To add on to what @aaronmgdr said, our dApp also works on localhost just fine but the same code doesn't work when hosted on our domains.

**Reply by glitch-txs** [reply]
Could you provide a minimal reproducible example?

@dckesler have you whitelisted your domain in the cloud website?

**Reply by dckesler** [reply]
Yes, we've tried whitelisting and also removing all Allowed Domains to see if that would help but so far we haven't been able to get past the `Failed to decrypt` error. Currently, we have the Allowed Domains empty.

Here is the repo, https://github.com/celo-org/staked-celo-web-app, it's pretty simple to run if you already have `yarn`.
- Clone repo
- Run `yarn`
- Run `yarn dev`

Then the first page at `localhost:3000` (or whatever port) will pop up a big `Connect Wallet` button. The local version should work totally fine connecting with for example https://react-wallet.walletconnect.com/walletconnect. (I've been going to Wallet Connect, copying the QR code value, and pasting it into the example wallet.) Then that same code is running at https://app.stcelo.xyz/connect and there it no longer works when trying to connect. This is using Next.js if that matters.

**Reply by glitch-txs** [reply]
it seems like you have the logic in a library instead of that repo?

**Reply by dckesler** [reply]
Here's maybe a better example for you that uses rainbowkit and wagmi instead of react-celo.
https://github.com/celo-org/staked-celo-web-app/blob/rainbowkit/src/pages/_app.tsx#L32

This is the same repo but on the rainbowkit branch. Then here is the hosted version of that branch https://staked-celo-web-app-git-rainbowkit-c-labs.vercel.app/connect

**Reply by nicolasbrugneaux** [reply]
Bumping this issue, as this is still blocking us from having WCv2 support on our website.

I've attached a screenshot of our attempt of using the WalletConnect modal library and the WalletConnect test wallet.

<img width="3168" alt="Screenshot 2023-07-03 at 17 20 22" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/3650621/59d5efb2-4037-4e8e-b1e5-c11e2a0f2fce">

**Reply by fnnhnsn** [reply]
@finessevanes, do you perhaps have any news on this? This is a critical issue and seems to block multiple projects from using mobile wallets via WalletConnect.

**Reply by glitch-txs** [reply]
@fnnhnsn this issue seems to be related to the wallet not supporting a specific chain, usually testnet. Could you show your chain configuration and maybe try with any other?

**Reply by fnnhnsn** [reply]
@glitch-txs right now we're just passing all supported chains, both testnet and mainnet, to the wagmi config and are handling those depending on the environment in the frontend. So like this:

```
const supportedChains = [mainnet, goerli, polygon, polygonMumbai, bsc, bscTestnet];

const { chains, publicClient } = configureChains<Chain>(
  supportedChains,
  [publicProvider(), ...ankrProvider, ...chainlistProvider, alchemyProvider({ apiKey })],
  {
    pollingInterval: 30_000,
    stallTimeout: 3000
  }
);
```

Should we minify this and try to just pass mainnet chains within that release candidate environment?

**Reply by glitch-txs** [reply]
There's something that wagmi does under the hood that's not that clear from the outside:

When you select the chains in the chains array like `[mainnet, goerli, bsc]` for example what wagmi does is, the first chain in the array will be set as a required chain in the session proposal, this means that if this first chain is not supported by the wallet the session connection will fail, the rest of chains in the array are going to be sent as optionalChains in the proposal which means that the wallet may or may not support them but the session can still be successful.

So mainly make sure the first chain in the array is one that the wallet supports, avoid testnets there.

Which wallet are you testing with btw?

**Reply by fnnhnsn** [reply]
Got it, will try that out on remote now, thanks. Testing with Metmask and Trust Wallet, which are both working just fine if I try them locally but are not working on a remote.

**Reply by glitch-txs** [reply]
@fnnhnsn that's weird, if the problem is still there please share a minimal reproducible example, I can try to reproduce it. For now I'd suggest trying to remove testnets.

Also you said you are using env vars for prod? Does this affect the config? Cz there could be a reason it's not working on production 🤔

**Reply by fnnhnsn** [reply]
Splitting that up did not help @glitch-txs , so just mentioning the mainnet chains as supported. Still a pending connection request and no answer.

**Reply by fnnhnsn** [reply]
You can check that e.g. [here](https://rc-app.polychainmonsters.com/collect/booster-packs), if you try to connect via WalletConnect with a Trust Wallet.

**Reply by glitch-txs** [reply]
@fnnhnsn it would be very useful if you can share a repo or a minimal example to check the code. So far I haven't been able to reproduce this with all WalletConnect examples:)

**Reply by fnnhnsn** [reply]
@glitch-txs  it's the same @onchainguy-eth posted above, so you could use that one. Is that sufficient? :) And have you tried our release candidate?

**Reply by fnnhnsn** [reply]
@glitch-txs  we created a minimal example here: https://wallet-connect-example-cac87bee70d1.herokuapp.com/
And this is the repo, feel free to work with it: https://github.com/polkamon/wallet-connect-example

**Reply by nicolasbrugneaux** [reply]
@fnnhnsn I don't know your current build situation, and I don't know how important performance is to you, but weirdly enough, removing the minification part of our webpack build solved everything. There must be an issue there.
- https://github.com/celo-org/staked-celo-web-app/pull/140

**Reply by glitch-txs** [reply]
Thanks @fnnhnsn your example works for me though, do you get the error on your side with this example?

Also, could you share your next.config.js file of your project?

**Reply by fnnhnsn** [reply]
@glitch-txs oh interesting, so you were able to connect a Trust Wallet via WalletConnect here? https://wallet-connect-example-cac87bee70d1.herokuapp.com/ I tried that with Trust Wallet on iOS, the latest version, and the confirmation within the wallet never popped up. 
The example has no individual next.config, this is the one of our original project:

```
const withNx = require('@nrwl/next/plugins/with-nx');
const webpackConfig = require('./webpack.config');
const { i18n } = require('./next-i18next.config');
const date = new Date();
const packages = require('./package.json');
const path = require('path');

/** @type {import("next").NextConfig} */
const nextConfig = {
  output: 'standalone',
  experimental: {
    outputFileTracingRoot: path.join(__dirname, '../../')
  },
  i18n,
  reactStrictMode: true,
  swcMinify: true,
  poweredByHeader: false,
  optimizeFonts: false,
  devIndicators: {
    buildActivityPosition: 'bottom-left'
  },
  images: {
    domains: [process.env.IMAGE_DOMAINS]
  },
  compiler: {
    removeConsole: process.env.NODE_ENV !== 'development' ? { exclude: ['error'] } : false
  },
  env: {
    NEXT_PUBLIC_BUILD_TIME: date.toString(),
    NEXT_PUBLIC_BUILD_TIMESTAMP: +date,
    NEXT_PUBLIC_APP_NAME: packages.name,
    NEXT_PUBLIC_APP_VERSION: '2.0.0'
  },
  //Based on https://nextjs.org/docs/advanced-features/security-headers
  async headers() {
    return [
      {
        source: '/:path*',
        headers: [
          {
            key: 'X-DNS-Prefetch-Control',
            value: 'on'
          },
          {
            key: 'Strict-Transport-Security',
            value: 'max-age=63072000; includeSubDomains; preload'
          },
          {
            key: 'X-XSS-Protection',
            value: '1; mode=block'
          },
          {
            key: 'X-Frame-Options',
            value: 'SAMEORIGIN'
          },
          {
            key: 'X-Content-Type-Options',
            value: 'nosniff'
          },
          {
            key: 'Referrer-Policy',
            value: 'origin-when-cross-origin'
          }
                 ]
      }
    ];
  },
  webpack: (config, options) => webpackConfig(config, options)
};

module.exports = withNx(nextConfig);

```

**Reply by glitch-txs** [reply]
Yes I was able to test with Trust Wallet and Rainbow, I was able to connect with both in Android.

Can you try removing swcMinify from next.config? Otherwise seems like what @nicolasbrugneaux shared could work as well, it's seems to be an issue with the bundler so far

**Reply by fnnhnsn** [reply]
Ah ok, that sounds like it might be an iOS issue. I'll try that today and will head back to you. Thanks a lot for your continuous support!

**Reply by fnnhnsn** [reply]
Turning off minification worked, thanks so much @glitch-txs @nicolasbrugneaux ! 🙏

**Comment by fnnhnsn** [comment]
> @fnnhnsn I don't know your current build situation, and I don't know how important performance is to you, but weirdly enough, removing the minification part of our webpack build solved everything. There must be an issue there.
> 
> * [Fix: stcelo v1 WC v2 fix attempts celo-org/staked-celo-web-app#140](https://github.com/celo-org/staked-celo-web-app/pull/140)


Thanks a lot, that's a great tip! Will take a look at that one.

**Reply by nicolasbrugneaux** [reply]
Further findings:

Disabling `swcMinify: false` (so still minifies, but uses babel minifier) also solves the issue.

**Reply by nicolasbrugneaux** [reply]
cc @glitch-txs, I think you can mark it as answered, maybe it'll help more people. We could also keep this open until we figure out _why_ [`swc`](https://swc.rs/) provokes that error

**Reply by glitch-txs** [reply]
@nicolasbrugneaux thanks! I see you're using next v12 is that right? [SWC seems to be set by default since v13](https://nextjs.org/docs/architecture/nextjs-compiler#minification), so maybe it could be related to an old issue that was fixed in the version update (just guessing).
